internal/interpreter/api.rs

```diff
@@ -1290,6 +1290,12 @@ impl ComponentInstance {
         d.set_property(comp.borrow(), &name, value)
     }
 
+    /// Set a callback triggered by `@debug-hook`-ed properties.
+    #[cfg(feature = "internal-highlight")]
+    pub fn set_debug_hook_callback(&self, observer: Option<crate::debug_hook::DebugHookCallback>) {
```

Comment:
This doesn't have to be an associated method of the `ComponentInstance` if it doesn't use `self`
caller can call `set_debug_hook_callback` durrectly.

That said, i'd rather not have a global for that and actually store it within `self`

Reply:
I was thinking the same:-)

But when I store it within the ComponentInstance: How do I find that again while evaluating my `DebugHook` expression? I have a `EvalLocalContext` there, but that has the instance of the component evaluating the expression only. Can I somehow conveniently get from there to the ComponentInstance I set my callback on? Do I need to propagate my callback over all component to have it available?

Reply:
Good question on where to put it.

I guess you can put it in the ComponentExtraData. Or in the ItemTreeDescription (but then this setter goes somewhere else)
Or maybe even in the compiler, and duplicate it from there.

Reply:
As suggested, the debug hook callback is now in `CompiledGlobalCollection`.